### PR TITLE
[Instagram] Extract video dimensions

### PR DESCRIPTION
**Pull request by TRox1972**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Extracts dimensions from Instagram videos.

### CODE REVIEW

File: youtube_dl/extractor/instagram.py [hunk 1]
```diff
@@ -102,8 +106,12 @@ def _real_extract(self, url):
                 like_count = int_or_none(media.get('likes', {}).get('count'))
                 comment_count = int_or_none(media.get('comments', {}).get('count'))
 
-        if not video_url:
-            video_url = self._og_search_video_url(webpage, secure=False)
+        if not formats:
+            formats = [{
+                'url': self._og_search_video_url(webpage, secure=False),
+                'height': int_or_none(media.get('dimensions', {}).get('height')),
+                'width': int_or_none(media.get('dimensions', {}).get('width')),
```
Comment (dstftw) [inline youtube_dl/extractor/instagram.py]: Code duplication.

File: youtube_dl/extractor/instagram.py [hunk 2]
```diff
@@ -93,7 +93,11 @@ def _real_extract(self, url):
             media = try_get(
                 shared_data, lambda x: x['entry_data']['PostPage'][0]['media'], dict)
             if media:
-                video_url = media.get('video_url')
+                formats = [{
+                    'url': media.get('video_url'),
+                    'height': int_or_none(media.get('dimensions', {}).get('height')),
+                    'width': int_or_none(media.get('dimensions', {}).get('width')),
```
Comment (dstftw) [inline youtube_dl/extractor/instagram.py]: Code duplication.

Comment (TRox1972) [inline youtube_dl/extractor/instagram.py]: fixed

File: youtube_dl/extractor/instagram.py [hunk 3]
```diff
@@ -102,8 +108,12 @@ def _real_extract(self, url):
...
                 comment_count = int_or_none(media.get('comments', {}).get('count'))
 
-        if not video_url:
-            video_url = self._og_search_video_url(webpage, secure=False)
+        if not formats.get('url'):
+            formats = [{
+                'url': self._og_search_video_url(webpage, secure=False),
+                'height': height,
+                'width': width,
+            }]
```
Comment (dstftw) [inline youtube_dl/extractor/instagram.py]: Nothing really changed. You construct the same structure two times.

Comment (TRox1972) [inline youtube_dl/extractor/instagram.py]: does it look better now?

Comment (dstftw) [inline youtube_dl/extractor/instagram.py]: Why not just
```
video_url = ...
width = ...
height = ...

...

if not video_url:
    video_url = ...

...

formats = [{
    'url': video_url,
    'width': width,
    'height': height,
}]
```

Comment (TRox1972) [inline youtube_dl/extractor/instagram.py]: That does seems better. Thanks, applied

File: youtube_dl/extractor/instagram.py [hunk 4]
```diff
@@ -105,6 +107,12 @@ def _real_extract(self, url):
         if not video_url:
             video_url = self._og_search_video_url(webpage, secure=False)
 
+        formats = [{
+            'url': video_url,
+            'width': width,
+            'height': height,
```
Comment (dstftw) [inline youtube_dl/extractor/instagram.py]: Dimensions are uninitialized when `media` is not available.

Comment (TRox1972) [inline youtube_dl/extractor/instagram.py]: fixed

**Comment by dstftw** [comment]
Cherry-picked in 16097822582b839a3744b54af90f7b3fd7132d26, thanks.

**Pull request closed without merging**